runtime/vm/ContinuationHelpers.cpp

```diff
@@ -89,8 +89,8 @@ createContinuation(J9VMThread *currentThread, j9object_t continuationObject)
 	return result;
 }
 
-void
-synchronizeWithConcurrentGCScan(J9VMThread *currentThread, J9VMContinuation *continuation)
+j9object_t
+synchronizeWithConcurrentGCScan(J9VMThread *currentThread, j9object_t continuationObject, J9VMContinuation *continuation)
 {
 	volatile uintptr_t *localAddr = &continuation->state;
 	/* atomically 'or' (not 'set') continuation->state  with currentThread */
```

Comment:
similarly to how I suggested helpers for setting/reseting concurrentScan we could have another helper setCarrierThreadToContinuationState (as we already getCarrierThreadFromContinuationState that hides some details how carrier ID is stored within the state), and then we would break bitOr into 2 steps:

uintptr_t newContinuationState = setCarrierThreadFromContinuationState(continuation->state, currentThread);
uintptr_t oldContinuationState = Atomic::set(&continuation->state, newContinuationState);

Reply:
the remaining comment deals with some details where we keep currentThread, so it should go inside this new helper (setCarrierThreadToContinuationState)

Reply:
let's use one helper to set both carrier ID and pending state (to hide also the details how we maintain pending state):

setCarrierThreadAndPendingStateToContinuationState

Reply:
last 2 comments have not been addressed